### PR TITLE
Rework Thread attachement event and unify implementation

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
#### Problem
GenericThreadStackManagerImpl_OpenThread,  when `CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT` is defined, is posting the connectivity establish event on the wrong condition causing the system to consider the thread attachment complete too soon and also causing multi-event posting.

#### Change overview

- Base the thread attachment establishment complet on the Thread device role and not the ipv6 interface up. The event can only be posted once per establishment state based on the Thread roles (Disable and Detached = Not attached. Child, Router and Leader= Attached). 
- There was also no reason to gate that logic behind `CHIP_SYSTEM_CONFIG_USE_OPEN_THREAD_ENDPOINT`
- Unify the use of the `GenericThreadStackManagerImpl_OpenThread` `_OnPlatformEvent`

I didn't want to mess in the ThreadStackManager Implementation of some platforms I don't know much about and that I can't test (Telink and Tizen), but the dev representing those platforms can now use the generic _OnPlatformEvent correctly. 

Linux and WebOs are separate as they use ot-br-possix apis.

#### Testing
Build and commission the EFR32 lighting-app

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp [hunk 1]
```diff
@@ -200,41 +200,38 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::_OnPlatformEvent(const
             }
         }
 #endif
-        Impl()->LockThreadStack();
```
Comment (jepenven-silabs) [inline src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp:203]: Why does the lock have to be removed? Is it safe to remove it when the Chip task can also interact with the threadInstance object?

File: src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h [hunk 2]
```diff
@@ -151,6 +151,7 @@ class GenericThreadStackManagerImpl_OpenThread
 
     otInstance * mOTInst;
     uint64_t mOverrunCount = 0;
+    bool mIsAttached       = 0;
```
Comment (jepenven-silabs) [inline src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h:154]: `= false` if we are dealing with a bool

**Review by jepenven-silabs — APPROVED** [review]

File: src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp (hunk 1 above)
Comment (jmartinez-silabs) [reply src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp:203]: The function _isThreadAttached lock around the thread api to get the lock. Rest of the functionally like posting the event or the log function (that does quite a lot of process for formatting strings}
) don't need to lock the thread instance.

File: src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h (hunk 2 above)
Comment (jmartinez-silabs) [reply src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.h:154]: Fair will change

**Review by LuDuda — APPROVED** [review]
Thanks :+1: LGTM

**Pull request merged**